components/script/dom/indexeddb/idbrequest.rs

```diff
@@ -46,6 +46,17 @@ struct RequestListener {
     iteration_param: Option<IterationParam>,
 }
 
+// https://w3c.github.io/IndexedDB/#transaction-lifecycle
+// A transaction is in this state after control returns to the event loop after its creation,
+// and when events are not being dispatched.”
+struct TxnActiveGuard<'a>(&'a IDBTransaction);
+
+impl Drop for TxnActiveGuard<'_> {
+    fn drop(&mut self) {
+        self.0.set_active_flag(false);
```

Comment:
So I don't think you need this, essentially you just need to set it to non-active after firing the event.